libraries/aws_redshift_cluster.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+require 'aws_backend'
+
+class AwsRedshiftCluster < AwsResourceBase
+  name 'aws_redshift_cluster'
+  desc 'Verifies an Redshift Cluster'
+
+  example "
+    describe aws_redshift_cluster(cluster_identifier: 'test-cluster-id') do
+      it { should exist }
+    end
+  "
+  def initialize(opts = {})
+    opts = { cluster_identifier: opts } if opts.is_a?(String)
+    super(opts)
+    validate_parameters(required: [:cluster_identifier])
+
+    raise ArgumentError, "#{@__resource_name__}: cluster_identifier must start with a letter followed by up to 62 letters/numbers/hyphens." if opts[:cluster_identifier] !~ /^[a-z]{1}[0-9a-z\-]{0,62}$/
+
+    catch_aws_errors do
+      @display_name = opts[:cluster_identifier]
+
+      begin
+        resp = @aws.redshift_client.describe_clusters(cluster_identifier: opts[:cluster_identifier])
+        return if resp.clusters.empty?
+        @rds_cluster = resp.clusters[0].to_h
+      rescue Aws::Redshift::Errors::NotAvailableValueError
```

Comment:
please confirm if this inherits from Aws::ServiceErrors and remove this rescue conditions